### PR TITLE
[Compute] vm create: Add provisionVMAgent configuration

### PR DESCRIPTION
**Pull request by qwordy**
[Compute] vm create: Add provisionVMAgent configuration

Implement #9757 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/vm/_template_builder.py [hunk 1]
```diff
@@ -278,6 +279,11 @@ def _build_os_profile():
                 }
             }
 
+        if provision_vm_agent is not None:
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/vm/_template_builder.py]: How about windows VM?

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 2]
```diff
@@ -292,6 +292,8 @@ def load_arguments(self, _):
             c.argument('eviction_policy', resource_type=ResourceType.MGMT_COMPUTE, min_api='2019-03-01',
                        arg_type=get_enum_type(VirtualMachineEvictionPolicyTypes, default=None),
                        help="The eviction policy for the low priority virtual machine.")
+        c.argument('provision_vm_agent', arg_type=get_three_state_flag(), min_api='2018-06-01',
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py]: How about enable_vm_agent?

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py]: Good name

File: src/azure-cli/azure/cli/command_modules/vm/_template_builder.py (hunk 1 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_template_builder.py]: Solved

**Pull request merged**